pkg/handlers/generic/mutation/taints/inject_worker.go

```diff
@@ -103,25 +109,25 @@ func (h *taintsWorkerPatchHandler) Mutate(
 
 	if err := patches.MutateIfApplicable(
 		obj, vars, &holderRef,
-		selectors.WorkersConfigTemplateSelector(eksbootstrapv1.GroupVersion.String(), "EKSConfigTemplate"), log,
-		func(obj *eksbootstrapv1.EKSConfigTemplate) error {
+		selectors.WorkersConfigTemplateSelector(eksbootstrapv1.GroupVersion.String(), "NodeadmConfigTemplate"), log,
+		func(obj *eksbootstrapv1.NodeadmConfigTemplate) error {
 			log.WithValues(
 				"patchedObjectKind", obj.GetObjectKind().GroupVersionKind().String(),
 				"patchedObjectName", ctrlclient.ObjectKeyFromObject(obj),
-			).Info("adding taints to worker node EKS config template")
-			if obj.Spec.Template.Spec.KubeletExtraArgs == nil {
-				obj.Spec.Template.Spec.KubeletExtraArgs = make(map[string]string, 1)
+			).Info("adding taints to worker NodeadmConfig template")
+			kubeletOptions := obj.Spec.Template.Spec.Kubelet
+			var runtimeConfigFromNodeadm *runtime.RawExtension
+			var flags []string
+			newTaints := toEKSConfigTaints(taintsVar)
+			if kubeletOptions != nil {
+				runtimeConfigFromNodeadm = kubeletOptions.Config
+				flags = kubeletOptions.Flags
 			}
-
-			existingTaintsFlagValue := obj.Spec.Template.Spec.KubeletExtraArgs["register-with-taints"]
-
-			newTaintsFlagValue := toEKSConfigTaints(taintsVar)
-
-			if existingTaintsFlagValue != "" {
-				newTaintsFlagValue = existingTaintsFlagValue + "," + newTaintsFlagValue
+			flags = append(flags, fmt.Sprintf("--register-with-taints=%s", newTaints))
+			obj.Spec.Template.Spec.Kubelet = &eksbootstrapv1.KubeletOptions{
+				Flags:  flags,
+				Config: runtimeConfigFromNodeadm,
 			}
```

Comment:
nit, maybe simplify this as below:

```suggestion
			newTaints := toEKSConfigTaints(taintsVar)
			kubeletOptions := ptr.Deref(obj.Spec.Template.Spec.Kubelet, &eksbootstrapv1.KubeletOptions{})
			kubeletOptions.Flags = append(kubeletOptions.Flags, fmt.Sprintf("--register-with-taints=%s", newTaints))
			obj.Spec.Template.Spec.Kubelet = kubeletOptions
```

Reply:
nice much simpler